trivy_report/report_generator.py

```diff
@@ -94,49 +120,86 @@ def parse_results(data: ReportDict, existing_issues: List[str]) -> Iterator[Repo
             raise TypeError(
                 f"The JSON entry .Results[{idx}] is not a dictionary, got: {type(result).__name__}"
             )
-        if "Vulnerabilities" not in result:
+
+        if "Vulnerabilities" in result:
+            vulnerabilities = result["Vulnerabilities"]
+            package_type = result["Type"]
+            for vulnerability in vulnerabilities:
+                package_name = vulnerability["PkgName"]
+                package_version = vulnerability["InstalledVersion"]
+                package_fixed_version = vulnerability["FixedVersion"]
+                package = f"{package_name}-{package_version}"
+                report_id = f"{package}"
+                has_issue = False
+                for existing_issue in existing_issues:
+                    issue_lower = existing_issue.lower()
+                    if (
+                        issue_lower.find(package_name.lower()) != -1
+                        and issue_lower.find(package_version.lower()) != -1
+                    ):
+                        has_issue = True
+                        break
+                if has_issue:
+                    continue
+
+                lookup_id = f"{package_type}:{report_id}"
+                report = reports.get(lookup_id)
+
+                if report is None:
+                    report = Report(
+                        kind="Vulnerability",
+                        id=report_id,
+                        package=package,
+                        package_name=package_name,
+                        package_version=package_version,
+                        package_fixed_version=package_fixed_version,
+                        package_type=package_type,
+                        target=result["Target"],
+                        vulnerabilities=[vulnerability],
+                    )
+                    reports[lookup_id] = report
+                else:
+                    report.vulnerabilities.append(vulnerability)
+            continue
+
+        if "Secrets" in result:
+            secrets = result["Secrets"]
+            for secret in secrets:
+                rule_id = secret["RuleID"]
+                category = secret["Category"]
+                startline = secret["StartLine"]
+                endline = secret["EndLine"]
+                has_issue = False
+                for existing_issue in existing_issues:
+                    issue_lower = existing_issue.lower()
+                    if (
+                        issue_lower.find(endline.lower()) != -1
+                        and issue_lower.find(startline.lower()) != -1
+                    ):
+                        has_issue = True
+                        break
+                if has_issue:
+                    continue
+
+                lookup_id = f"{startline}:{endline}"
+                report = reports.get(lookup_id)
+
+                if report is None:
+                    report = Report(
```

Comment:
It's probably a better idea to have two separate report types, which are better suited to that specific type.